storage/gcloud/aio/storage/storage.py

```diff
@@ -141,17 +120,25 @@ def __init__(self, storage, name):
 
         return Blob(self, blob_name, content)
 
-    def new_blob(self, blob_name):
+    async def list_blobs(self, prefix='', session=None):
+        params = {'prefix': prefix}
 
-        blob = Blob(self, blob_name, {'size': 0})
+        status, content = await self.storage.list_objects(self.name,
+                                                          params=params,
+                                                          session=session)
 
-        return blob
+        if status < 200 or status >= 300:
+            log.error('Could not list %s/%s: %s', self.name, prefix, content)
+            return
 
+        return [x['name'] for x in content.get('items', list())]
```

Comment:
Do you have to `content = ujson.loads(content)`?

Reply:
Nope, calling the `get` in the `Storage` method with `json_response=True` already does this.